pkgs/top-level/haskell-ng.nix

```diff
@@ -16,7 +17,10 @@ rec {
     ghc722 = callPackage ../development/compilers/ghc/7.2.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc742 = callPackage ../development/compilers/ghc/7.4.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc763 = callPackage ../development/compilers/ghc/7.6.3.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
-    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix { ghc = compiler.ghc742Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
+    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix {
+      ghc = if pkgs.stdenv.isDarwin then compiler.ghc783Binary else compiler.ghc742Binary;
```

Comment:
FWIW, here's a log of building ghc784 with the 742Binary on darwin.

https://gist.github.com/gridaphobe/6c340b84d70780e9121f

Perhaps we could instruct ghc742Binary to use `cpphs` as a pre-processor instead of adding ghc783Binary?